frontend/packages/topology/src/components/dropdowns/ApplicationSelector.tsx

```diff
@@ -77,6 +79,12 @@ const ApplicationSelector: React.FC<ApplicationSelectorProps> = ({
     ? t('topology~Warning: the Application grouping already exists.')
     : t('topology~A unique name given to the Application grouping to label your resources.');
 
+  React.useEffect(() => {
+    if (selectedKey.value === CREATE_APPLICATION_KEY && nameField?.value === '') {
+      applicationNameInputRef.current?.focus();
+    }
+  }, [nameField, selectedKey.value]);
+
```

Comment:
The application name is always empty when switching between "New application" and an existing application. But also when the field input wouldn't be cleared when selecting an existing application: Is there any reason why the user don't want jump to the input field if the user selects "Create application"? I don't see a reason for the second condition here.

Isn't this easier and matches still what you would expect? Wdyt?

```suggestion
  React.useEffect(() => {
    if (selectedKey.value === CREATE_APPLICATION_KEY) {
      applicationNameInputRef.current?.focus();
    }
  }, [selectedKey.value]);
```